reader/static/js/init.js

```diff
@@ -2676,6 +2680,61 @@ function UI_Loda_Search(o) {
 
 }
 
+function UI_Loda_Jump(o) {
+	o=be(o);
+	UI_Loda.call(this, {
+		node: o.node,
+		kind: ['Loda_Jump'].concat(o.kind || []),
+		name: 'Jump',
+		html: o.html || `<div class="Loda-window" tabindex="-1"><header data-bind="header"></header><button class="ico-btn close" data-bind="close"></button><content data-bind="content">
+				<div class="Jump-Wrapper">
+					<input type="number" data-bind="input_chap" placeholder="Chap." />
+					<input type="number" data-bind="input_page" placeholder="Page" />
+					<button class="Jump-Btn" data-bind="btn"></button>
+				</div>
+			</content></div>`
+	});
+	this.manager = o.manager;
+	this.name = 'Jumper';
+	this.noPropagation = true;
+	this.focusElement = this._.input_chap;
+
+	this.btn = new UI_Button({
+		node: this._.btn,
+		text: 'Go'
+	});
+
+	this.input_chap = new UI_Input({
+		node: this._.input_chap
+		})
+
+	this.input_page = new UI_Input({
+		node: this._.input_page
+		})
+
+	this.jump = () => {
+		let chap = this._.input_chap.value, page = this._.input_page.value || 1;
+		try {
+			if (page > 40) throw err
```

Comment:
I'm not the biggest fan of having this magic number here, especially since there can be some proxy chapters with more than 40 pages (the first chapter of OnK comes to mind). You can grab the page count from the `Reader.SCP` object; you'll just have to make some small changes to manage the lifecycle of retrieving it. 

I'm also a bit undecided on whether we need to throw here. You can just short circuit it with an if/else since this is the only thing in this block that'll throw anyway.

Reply:
I used throw there because, that block can throw different errors too, an invalid chapter number will throw an error, so I wanted to make it consistent so invalid page number throws an error too, even if we do if/else, we'll have to use the try block.
Regarding that 40, I asked appu before doing it, I'll see Reader.SCP object too. Probably I had in mind that this is working only with guya.

Reply:
Ah that's a good point, I suppose you'll get an error down the line. It might be better to validate those inputs before initializing the chapter though since you'll have side-effects on the failure.

Also, nitpick but it might be better to throw an apt error message. Right now it'd fail on err not being defined.

Reply:
Alrighty, thanks